### ISSUE TITLE
Allow for extensibility in pDataDT

### ISSUE THREAD

**Issue by NicoleEO** [giotto-suite/Giotto#83]
Currently pDataDT does checks for specific class types, but does not allow use of classes that have been derived from these. Can the following edits be made to allow for inheritance of these classes?
![image](https://user-images.githubusercontent.com/20404635/97621226-e1b3f200-19df-11eb-9bca-41b20a52d2fc.png)

**Comment by RubD** [comment]
Hi Nicole,

I agree that this is indeed a better way. You can make a PR or I can fix it myself. This is actually a function of one of the early days of Giotto. We still have quite some work to do to make Giotto interoperable with other classes and packages.

Thanks!
Ruben

**Comment by RubD** [comment]
I've updated the code and I'll close the comment. Thanks again!

**Comment by NicoleEO** [comment]
Thanks Ruben!

**Comment by NicoleEO** [comment]
Hi Ruben, 

Out of curiosity, are you planning on submitting this to Bioconductor or CRAN in the near future?

Thanks,
Nicole

**Comment by RubD** [comment]
Hi Nicole,

That’s a good question. We have worked towards submission on CRAN and everything should be CRAN proof. However at the last moment we noticed that CRAN does not allow you to submit a package if it uses other packages in development. Since many of these packages are specific to spatial analyses (e.g. HMRF, trendsceek, SPARK, and others in the future) we are currently figuring out what the best way forward is. We want to make it as easy as possible to install, but also provide users with all the latest spatial tools.

We are also thinking about Bioconductor, but in that case we would have to (or like to) change some of the package data structures so that it is compatible with other bioconducter packages and classes. 

If you have specific ideas or would like to contribute, then you’re more than welcome to do so.

Best,
Ruben
